src/app/signup/signup.component.html

```diff
@@ -2,13 +2,21 @@ <h3>Signup</h3>
 
 <main>
   <form [formGroup]="form" (ngSubmit)="signup()">
-    <label for="email">Email address</label>
-    <input type="email" id="email" formControlName="email" required>
-    <label for="password">Password</label>
-    <input type="password" id="password" formControlName="password" required>
-    <label for="confirm">Confirm Password</label>
-    <input type="password" id="confirm" formControlName="confirm" required>
-    <button type="submit">Login</button>
+    <mat-form-field>
+      <mat-label for="email">Email address</mat-label>
+      <input matInput type="email" id="email" formControlName="email" required>
+      <mat-error><p>Please enter a valid email</p></mat-error>
+    </mat-form-field>
+    <mat-form-field>
+      <mat-label for="password">Password</mat-label>
+      <input matInput type="password" id="password" formControlName="password" required>
```

Comment:
add mat-error to display a message is password is not filled.

Reply:
Done